### PR TITLE
kubectl/describe: prefix array elements for readability

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Prefix array elements with dash if array contains multiple elements and at least one object.

This should disambiguate display of CRD's:
```yaml
items:
  - foo: a
    bar: b
```
and
```yaml
items:
  - foo: a
  - bar: b
```

#### Does this PR introduce a user-facing change?

```release-note
kubectl describe prefixes array elements with dash if array contains multiple elements and at least one object.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/describe/describe_test.go [hunk 1]
```diff
@@ -4668,22 +4699,86 @@ URL:	http://localhost
...
 		},
 	}
-	printUnstructuredContent(w, LEVEL_0, obj.UnstructuredContent(), "", ".dummy1", ".metadata.dummy3")
+	printUnstructuredContent(w, LEVEL_0, obj.UnstructuredContent(), false, "", ".dummy1", ".metadata.dummy3")
 	output := out.String()
 
 	for _, test := range testCases {
 		if len(test.expected) > 0 {
 			if !strings.Contains(output, test.expected) {
+				t.Logf("%s", output)
```
Comment (szuecs) [inline staging/src/k8s.io/kubectl/pkg/describe/describe_test.go:4781]: You don't need this, it's already printed below in %q

Comment (AlexanderYastrebov) [reply staging/src/k8s.io/kubectl/pkg/describe/describe_test.go:4781]: This prints multipline output while Errorf below escapes new lines due to %q

Comment (szuecs) [reply staging/src/k8s.io/kubectl/pkg/describe/describe_test.go:4781]: Yes but do we need this?
You verified the output and then we can drop the output.

**Pull request closed without merging**